cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,636 @@
+            "Time Elapsed": self.migration_request_details_list.get_clock(vm_name),
+        }
+        return status
+
+    def filter_by_status(self, status):
+        """Set filter_by_dropdown to 'Status' and uses status arg by user to set status filter.
+
+        Args:
+            status(str): Takes status string as arg. Valid status options are:
+                         ['Pending', 'Validating', 'Pre-migration', 'Migrating',
+                          'VM Transformations Ccompleted', 'VM Transformations Failed']
+        """
+        try:
+            self.filter_by_dropdown.item_select("Status")
+            self.filter_by_status_dropdown.item_select(status)
+        except NoSuchElementException:
+            raise ItemNotFound(
+                "Migration plan is in Not Started State,"
+                " hence filter status dropdown not visible"
+            )
+
+    def sort_by(self, option):
+        """Sort VM list by using one of the 'Started','VM Name' or 'Status' option.
+
+        Args:
+            option(str): Takes option string as arg.
+        """
+        try:
+            self.sort_by_dropdown.item_select(option)
+        except NoSuchElementException:
+            raise ItemNotFound(
+                "Migration plan is in Not Started State," " hence sort_by dropdown not visible"
+            )
+
+    def plan_in_progress(self, vms_count=5):
+        """Reuturn True or False, depending on migration plan status.
+
+        If none of the VM migrations are in progress, return True.
+        """
+        self.items_on_page.item_select("15")
+        migration_plan_in_progress_tracker = []
+        vms = self.migration_request_details_list.read()
+        for vm in vms:
+            clock_reading1 = self.migration_request_details_list.get_clock(vm)
+            time.sleep(1)  # wait 1 sec to see if clock is ticking
+            logger.info(
+                "For vm %s, current message is %s",
+                vm,
+                self.migration_request_details_list.get_message_text(vm),
+            )
+            logger.info(
+                "For vm %s, current progress description is %s",
+                vm,
+                self.migration_request_details_list.get_progress_description(vm),
+            )
+            clock_reading2 = self.migration_request_details_list.get_clock(vm)
+            logger.info("clock_reading1: %s, clock_reading2:%s", clock_reading1, clock_reading2)
+            logger.info(
+                "For vm %s, is currently in progress: %s",
+                vm,
+                self.migration_request_details_list.is_in_progress(vm),
+            )
+            migration_plan_in_progress_tracker.append(
+                self.migration_request_details_list.is_in_progress(vm)
+                and (clock_reading1 < clock_reading2)
+            )
+        return not any(migration_plan_in_progress_tracker)
+
+
+@attr.s
+class MigrationPlan(BaseEntity):
+    """Class representing v2v Migration Plan"""
```

Comment:
Docblock defining the attributes for the entity.  What is csv_import, a string, a filename, a dictionary?  Is target_provider a Provider entity?  Is osp_security_group and osp_flavor strings or entities?  Are pre_playbook and post_playbook strings?